site-setup/learn_article.ftl

```diff
@@ -184,181 +167,195 @@ html{box-sizing:border-box;-ms-overflow-style:scrollbar}*,::after,::before{box-s
-		const breadcrumbCurrentArticle = document.getElementById('breadcrumbCurrentArticle');
+				targets.forEach(target => {
+					const node = target.id ? document.getElementById(target.id) : null;
 
-		const productDocumentationSelector = document.getElementById('productDocumentationSelector');
+					if (node) {
+						observer.observe(node);
 
-		if (breadcrumbCurrentArticle && productDocumentationSelector) {
+						node.style.cssText = "margin-top: -157px; padding-top: 157px;"
+					}
+				});
 
-			// Preselect documentation dropdown to be the current article's product
+				// Documentation dropdown
 
-			const currentDocProduct = breadcrumbAncestor ? breadcrumbAncestor : breadcrumbCurrentArticle;
+				const productDocumentationSelector = document.getElementById('productDocumentationSelector');
 
-			const currentDocProductValue = currentDocProduct.innerText.replace(/\s/g, '-').toLowerCase();
+				if (productDocumentationSelector) {
+					// Preselect documentation dropdown to be the current article's product
 
-			productDocumentationSelector.value = currentDocProductValue;
+					const breadcrumbFirstEntry = document.querySelector('.breadcrumb-wrapper a');
 
-			// Route user to the selected documentation landing page
+					const pathName = window.location.pathname;
 
-			productDocumentationSelector.addEventListener('change', event => {
-				const target = event.target.value;
+					if (breadcrumbFirstEntry) {
+						const breadcrumbFirstEntryValue = breadcrumbFirstEntry.innerText.replace(/\s/g, '-').toLowerCase();
 
-				if (target !== currentDocProductValue) {
-					if (target === 'analytics-cloud') {
-						window.location.pathname = '/analytics-cloud/latest/en/index.html';
-					} else if (target === 'commerce') {
-						window.location.pathname = '/commerce/latest/en/index.html';
-					} else if (target === 'dxp-cloud') {
-						window.location.pathname = '/dxp-cloud/latest/en/index.html';
-					} else if (target === 'reference') {
-						window.location.pathname = '/reference/latest/en/index.html';
+						productDocumentationSelector.value = breadcrumbFirstEntryValue;
+					}
+					else if (pathName.includes("analytics-cloud")) { // if breadcrumb is empty, then defer to URL
+						productDocumentationSelector.value = "analytics-cloud";
+					} else if (pathName.includes("commerce")) {
+						productDocumentationSelector.value = "commerce";
+					} else if (pathName.includes("dxp-cloud")) {
+						productDocumentationSelector.value = "dxp-cloud";
+					} else if (pathName.includes("reference")) {
+						productDocumentationSelector.value = "reference";
 					} else {
-						window.location.pathname = '/dxp/latest/en/index.html';
+						productDocumentationSelector.value = "dxp";
 					}
+
+					// Route user to the selected documentation landing page
+
+					productDocumentationSelector.addEventListener('change', event => {
+						const target = event.target.value;
```

Comment:
Can you look into whether we can build the dropdown menu with a navigation menu? Then we wouldn't have to hardcode the paths